File: app/src/main/java/com/elseboot3909/GCRClient/Utils/AccountUtils.java
package com.elseboot3909.GCRClient.Utils;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import androidx.core.graphics.drawable.RoundedBitmapDrawable;
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory;

import com.elseboot3909.GCRClient.Entities.AvatarInfo;
import com.elseboot3909.GCRClient.R;
import com.google.android.material.chip.Chip;
import com.squareup.picasso.Picasso;
import com.squareup.picasso.Target;

import java.util.Random;

public class AccountUtils {

    static final Random random = new Random();

    public static final int[] dummyAvatars = new int[]{R.drawable.ic_dummy_avatar_1, R.drawable.ic_dummy_avatar_2, R.drawable.ic_dummy_avatar_3, R.drawable.ic_dummy_avatar_4, R.drawable.ic_dummy_avatar_5};

    public static int getRandomAvatar() {
        return dummyAvatars[random.nextInt(dummyAvatars.length)];
    }

    public static final String[] dummyUsernames = new String[]{"John", "Laurel", "Luke", "Jack", "Dexter", "Henry", "Dale", "Elbert"};

    public static String getRandomUsername() {
        return dummyUsernames[random.nextInt(dummyUsernames.length)];
    }

    public static void setAvatarDrawable(AvatarInfo avatarInfo, View view)  {
        Picasso picasso = Picasso.get();
        picasso.load(avatarInfo.getUrl()).into(new Target() {
            @Override
            public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
                RoundedBitmapDrawable roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(Resources.getSystem(), bitmap);
                roundedBitmapDrawable.setCircular(true);
                if (view instanceof Chip) {
                    ((Chip) view).setChipIcon(roundedBitmapDrawable);
                } else if (view instanceof ImageView) {
                    ((ImageView) view).setImageDrawable(roundedBitmapDrawable);
                } else {
                    Log.e(Constants.LOG_TAG, "(" + this.getClass() + ") setAvatarDrawable: Check type of view object!");
                }
            }

            @Override
            public void onBitmapFailed(Exception e, Drawable errorDrawable) { }

            @Override
            public void onPrepareLoad(Drawable placeHolderDrawable) { }
        });
    }

}

File: app/src/main/java/com/elseboot3909/GCRClient/Utils/AccountUtils.kt
package com.elseboot3909.GCRClient.Utils

import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.view.View
import android.widget.ImageView
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory
import com.elseboot3909.GCRClient.Entities.AvatarInfo
import com.elseboot3909.GCRClient.R
import com.google.android.material.chip.Chip
import com.squareup.picasso.Picasso
import com.squareup.picasso.Target
import java.util.*

class AccountUtils {

    companion object {

        private val random = Random()

        private val dummyAvatars = arrayOf(R.drawable.ic_dummy_avatar_1, R.drawable.ic_dummy_avatar_2, R.drawable.ic_dummy_avatar_3, R.drawable.ic_dummy_avatar_4, R.drawable.ic_dummy_avatar_5)

        fun getRandomAvatar() : Int {
            return dummyAvatars[random.nextInt(dummyAvatars.size)]
        }

        private val dummyUsernames = arrayOf("John", "Laurel", "Luke", "Jack", "Dexter", "Henry", "Dale", "Elbert")

        fun getRandomUsername() : String {
            return dummyUsernames[random.nextInt(dummyUsernames.size)]
        }

        fun setAvatarDrawable(avatarInfo: AvatarInfo , view: View)
        {
            val picasso = Picasso.get()
            picasso.load(avatarInfo.url).into(object: Target {
                override fun onBitmapLoaded(bitmap: Bitmap?, from: Picasso.LoadedFrom?) {
                    val roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(Resources.getSystem(), bitmap)
                    roundedBitmapDrawable.isCircular = true
                    if (view is Chip) {
                        view.chipIcon = roundedBitmapDrawable
                    } else if (view is ImageView) {
                        view.setImageDrawable(roundedBitmapDrawable)
                    }
                }

                override fun onBitmapFailed(e: java.lang.Exception?, errorDrawable: Drawable?) { }

                override fun onPrepareLoad(placeHolderDrawable: Drawable?) { }

            })
        }
    }
}
